i6_models/config.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import MISSING, dataclass
```

Comment:
I would add some docstring explaining (high-level) what the logic of the config would be.

Reply:
Do you want an example (e.g. one of the classes from the tests)? Because for now this is not really set I think. For now we wanted a base definition model configurations are built upon.

Reply:
But you must have some basic understanding of how this model config would be used, or not?

Reply:
It would also be helpful to document the inspiration from Fairseq.

Reply:
> But you must have some basic understanding of how this model config would be used

Yes, construction for example as in the tests. But for example then the later use for models and integration into our pipeline is not fully decided from my understanding. What I was asking was, whether you want an example or a rather conceptual explaining here.

Reply:
If possible, both, a conceptual explaining, and a minimal example.
I'm fine with the most minimal (1 line of code) example you can think of.
The conceptual explanation can also be short for now (one sentence).
A reference to the Fairseq code would also be very helpful.

Reply:
I also mean a costring for the module itself.

Reply:
So how the config then later is used in the module? This is highly up to the user and also not set yet. 
The idea discussed in the meeting was to provide a base config class similar to Fairseq, which can then be used to build the first modules. But this can be added later once we have a "common" way of using these configs.

Reply:
I'm fine with whatever minimal example you can think of, and whatever most minimal explanation you can give.

Reply:
Looks good!